kuma/wiki/helpers.py

```diff
@@ -355,3 +355,12 @@ def wiki_url(context, path):
     # finally cache the reversed document URL for a bit
     memcache.set(u'wiki_url:%s:%s' % (locale, path), url, 60 * 5)
     return url
+
+
+@register.function
+def addbreaks(title):
+    """
+    adds <wbr> tags before . that appear between letters
+    no numbers should be split
+    """
+    return re.sub(r"(?<=[a-z,A-Z])\.(?=[a-z,A-Z])", "<wbr>.", title)
```

Comment:
That looks sensible, except that we should separate the compilation of the regex pattern and the call to sub into two different pieces. What happens when you call `re.sub` is that it takes the raw string you pass and then compiles it into a pattern object, which can be time costly. If you instead compile the pattern manually on the module level (outside the function), it'll only do that compilation once per server process instead on every call. So this should be:
```python
BREAKS_RE = re.compile(r'(?<=[a-z,A-Z])\.(?=[a-z,A-Z])')

def addbreaks(title):
    """
    adds <wbr> tags before . that appear between letters
    no numbers should be split
    """
    return BREAKS_RE.sub('<wbr>.', title)
```
Please put `BREAKS_RE` to the top of the module.